app/src/lib/editors/darwin.ts

```diff
@@ -1,174 +1,135 @@
-): string {
-  switch (editor) {
-    case ExternalEditor.Atom:
-      return Path.join(installPath, 'Contents', 'Resources', 'app', 'atom.sh')
-    case ExternalEditor.VSCode:
-    case ExternalEditor.VSCodeInsiders:
-      return Path.join(
-        installPath,
-        'Contents',
-        'Resources',
-        'app',
-        'bin',
-        'code'
-      )
-    case ExternalEditor.VSCodium:
-      return Path.join(
-        installPath,
-        'Contents',
-        'Resources',
-        'app',
-        'bin',
-        'code'
-      )
-    case ExternalEditor.MacVim:
-      return Path.join(installPath, 'Contents', 'MacOS', 'MacVim')
-    case ExternalEditor.SublimeText:
-      return Path.join(installPath, 'Contents', 'SharedSupport', 'bin', 'subl')
-    case ExternalEditor.BBEdit:
-      return Path.join(installPath, 'Contents', 'Helpers', 'bbedit_tool')
-    case ExternalEditor.PhpStorm:
-      return Path.join(installPath, 'Contents', 'MacOS', 'phpstorm')
-    case ExternalEditor.PyCharm:
-      return Path.join(installPath, 'Contents', 'MacOS', 'pycharm')
-    case ExternalEditor.RubyMine:
-      return Path.join(installPath, 'Contents', 'MacOS', 'rubymine')
-    case ExternalEditor.TextMate:
-      return Path.join(installPath, 'Contents', 'Resources', 'mate')
-    case ExternalEditor.Brackets:
-      return Path.join(installPath, 'Contents', 'MacOS', 'Brackets')
-    case ExternalEditor.WebStorm:
-      return Path.join(installPath, 'Contents', 'MacOS', 'WebStorm')
-    case ExternalEditor.IntelliJ:
-      return Path.join(installPath, 'Contents', 'MacOS', 'idea')
-    case ExternalEditor.Typora:
-      return Path.join(installPath, 'Contents', 'MacOS', 'Typora')
-    case ExternalEditor.CodeRunner:
-      return Path.join(installPath, 'Contents', 'MacOS', 'CodeRunner')
-    case ExternalEditor.SlickEdit:
-      return Path.join(installPath, 'Contents', 'MacOS', 'vs')
-    case ExternalEditor.Xcode:
-      return '/usr/bin/xed'
-    case ExternalEditor.GoLand:
-      return Path.join(installPath, 'Contents', 'MacOS', 'goland')
-    case ExternalEditor.AndroidStudio:
-      return Path.join(installPath, 'Contents', 'MacOS', 'studio')
-    case ExternalEditor.Rider:
-      return Path.join(installPath, 'Contents', 'MacOS', 'rider')
-    case ExternalEditor.Nova:
-      return Path.join(installPath, 'Contents', 'SharedSupport', 'nova')
-    default:
-      return assertNever(editor, `Unknown external editor: ${editor}`)
-  }
+  /**
+   * List of bundle identifiers that are used by the app in its multiple
+   * versions.
+   **/
+  readonly bundleIdentifiers: string[]
 }
 
-async function findApplication(editor: ExternalEditor): Promise<string | null> {
-  const identifiers = getBundleIdentifiers(editor)
-  for (const identifier of identifiers) {
+/**
+ * This list contains all the external editors supported on macOS. Add a new
+ * entry here to add support for your favorite editor.
+ **/
+const editors: IDarwinExternalEditor[] = [
```

Comment:
Given how simple this type is you could simplify it to be a Record type keyed on the name if you wanted.  Something like

```ts
const editors: Record<string, ReadonlyArray<string>> = {
  Atom: ['com.github.atom'],
  MacVim: ['org.vim.MacVim'],
  'Visual Studio Code': ['com.microsoft.VSCode'],
  'Visual Studio Code (Insiders)': ['com.microsoft.VSCodeInsiders'],
  VSCodium: ['com.visualstudio.code.oss'],
  'Sublime Text': [
    'com.sublimetext.4',
    'com.sublimetext.3',
    'com.sublimetext.2',
  ],
  BBEdit: ['com.barebones.bbedit'],
  PhpStorm: ['com.jetbrains.PhpStorm'],
  PyCharm: ['com.jetbrains.PyCharm'],
  RubyMine: ['com.jetbrains.RubyMine'],
  TextMate: ['com.macromates.TextMate'],
  Brackets: ['io.brackets.appshell'],
  WebStorm: ['com.jetbrains.WebStorm'],
  Typora: ['abnerworks.Typora'],
  CodeRunner: ['com.krill.CodeRunner'],
  SlickEdit: [
    'com.slickedit.SlickEditPro2018',
    'com.slickedit.SlickEditPro2017',
    'com.slickedit.SlickEditPro2016',
    'com.slickedit.SlickEditPro2015',
  ],
  IntelliJ: ['com.jetbrains.intellij'],
  Xcode: ['com.apple.dt.Xcode'],
  GoLand: ['com.jetbrains.goland'],
  'Android Studio': ['com.google.android.studio'],
  Rider: ['com.jetbrains.rider'],
  Nova: ['com.panic.Nova'],
}
```

Can't get much more compact than that eh? Totally up to you. Then you can use `Object.entries` to iterate over them

```ts
  for (const [editor, bundleIdentifiers] of Object.entries(editors)) {
```

Reply:
That's a good one! 🤩 

However I'm tempted to leave the current approach I have 😳 , since I think it makes the code (and the documentation in `editor-integration.md`) more consistent with Linux and Windows. What do you think?

Reply:
Fine by me!